packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -146,50 +218,106 @@ const useRootStyles = makeStyles({
+  border: '0',
+  padding: '0',
+  color: tokens.colorNeutralForeground1,
+  backgroundColor: 'transparent',
+  ...typographyStyles.body1,
+
+  '::placeholder': {
+    color: tokens.colorNeutralForeground4,
+    opacity: 1, // browser style override
+  },
+});
+
 const useInputStyles = makeStyles({
-  base: {
-    gridColumnStart: '1',
-    gridColumnEnd: '2',
-    gridRowStart: '1',
-    gridRowEnd: '3',
-    outlineStyle: 'none',
-    ...shorthands.padding(0),
+  small: {
+    ...typographyStyles.caption1,
+  },
+  disabled: {
+    color: tokens.colorNeutralForegroundDisabled,
+    cursor: 'not-allowed',
+    backgroundColor: tokens.colorTransparentBackground,
+    '::placeholder': {
+      color: tokens.colorNeutralForegroundDisabled,
+    },
   },
 });
 
-const useButtonStyles = makeStyles({
-  base: {
-    display: 'inline-flex',
-    width: '24px',
-    alignItems: 'center',
-    justifyContent: 'center',
-    ...shorthands.border(0),
-    position: 'absolute',
+const buttonBaseStyles: GriffelResetStyle = {
+  display: 'inline-flex',
+  width: '24px',
+  alignItems: 'center',
+  justifyContent: 'center',
+  border: '0',
+  position: 'absolute',
 
+  outlineStyle: 'none',
+  height: '100%',
+
+  backgroundColor: 'transparent',
+  color: tokens.colorNeutralForeground3,
+  ':active': {
     outlineStyle: 'none',
-    height: '100%',
+  },
 
-    ':enabled:hover': {
+  ':enabled': {
+    ':hover': {
       cursor: 'pointer',
+      color: tokens.colorNeutralForeground3Hover,
+      backgroundColor: tokens.colorSubtleBackgroundHover,
     },
-
     ':active': {
-      outlineStyle: 'none',
+      color: tokens.colorNeutralForeground3Pressed,
+      backgroundColor: tokens.colorSubtleBackgroundPressed,
     },
-
-    ':disabled': {
-      cursor: 'not-allowed',
+    [`&.${spinButtonExtraClassNames.buttonActive}`]: {
+      color: tokens.colorNeutralForeground3Pressed,
+      backgroundColor: tokens.colorSubtleBackgroundPressed,
     },
   },
 
-  incrementButton: {
-    gridColumnStart: '2',
-    gridColumnEnd: '3',
-    gridRowStart: '1',
-    gridRowEnd: '2',
-    ...shorthands.borderRadius(0, tokens.borderRadiusMedium, 0, 0),
+  ':disabled': {
+    cursor: 'not-allowed',
+    color: tokens.colorNeutralForegroundDisabled,
   },
+};
+
+const incrementButtonBaseStyles: GriffelResetStyle = {
+  gridColumnStart: '2',
+  gridColumnEnd: '3',
+  gridRowStart: '1',
+  gridRowEnd: '2',
+  borderRadius: `0 ${tokens.borderRadiusMedium} 0 0`,
+  padding: '4px 5px 1px 5px',
+};
+
+const decrementButtonBaseStyles: GriffelResetStyle = {
+  gridColumnStart: '2',
+  gridColumnEnd: '3',
+  gridRowStart: '2',
+  gridRowEnd: '3',
+  borderRadius: `0 0 ${tokens.borderRadiusMedium} 0`,
+  padding: '1px 5px 4px 5px',
+};
+
+const useIncrementButtonClassName = makeResetStyles({
+  ...buttonBaseStyles,
+  ...incrementButtonBaseStyles,
+});
 
+const useDecrementButtonClassName = makeResetStyles({
+  ...buttonBaseStyles,
+  ...decrementButtonBaseStyles,
+});
```

Comment:
This results in all of `buttonBaseStyles` being emitted in the bundle twice, one for each class. Probably not a huge deal since it'll likely be gzipped, but it still might be better to put as much as possible that's shared into a single class, and then individual `makeStyles` classes to override the few props that are different.

E.g. the base class could include:
```
  gridColumn: '2',
  borderRadius: 0,
  padding: '1px 5px',
```  

And then each of the individual buttons would override, e.g.:
```
increment: {
  gridRowStart: '1',
  borderTopRightRadius: tokens.borderRadiusMedium,
  paddingTop: '4px',
},
decrement: {
  gridRowStart: '2',
  borderBottomRightRadius: tokens.borderRadiusMedium,
  paddingBottom: '4px', // (might not be needed?)
},
```

`gridRowEnd` shouldn't be needed, because its value of `auto` should resolve to the correct value to span 1 row.